src/prefect/agent/agent.py

```diff
@@ -679,7 +674,14 @@ def _get_flow_run_metadata(
             }
         }
         result = self.client.graphql(query)
-        return result.data.flow_run
+        return sorted(
+            result.data.flow_run,
+            key=lambda flow_run: getattr(
+                StateSchema().load(flow_run.serialized_state),
+                "start_time",
+                pendulum.now("utc"),
+            ),
```

Comment:
Could this be `key=lambda flow_run: flow_run.serialized_state.get("start_time", pendulum.now("utc").isoformat())` ? I feel like we are de-serializing the state a bunch of times.